api/src/opentrons/protocol_api_experimental/well.py

```diff
@@ -101,10 +103,17 @@ def from_center_cartesian(  # noqa: D102
         raise NotImplementedError()
 
     def __repr__(self) -> str:  # noqa: D105
-        raise NotImplementedError()
+        # TODO: (spp, 2021.07.14): Should this be a combination of display name & <obj>?
+        return f"Well:{self._well_name}<Labware:{self.parent}>"
 
-    def __eq__(self, other: object) -> bool:  # noqa: D105
-        raise NotImplementedError()
+    def __eq__(self, other: object) -> bool:
+        """Compare for object equality.
+
+        Checks that other object is a `Well` and belongs to the same labware.
+        """
+        return (isinstance(other, Well) and
+                self.well_name == other.well_name and
+                self.parent is other.parent)
```

Comment:
I think this should be a value equality check, not reference equality:

```suggestion
                self.parent == other.parent)
```

Not a change request, but just food for thought: this really makes me want to ditch the `parent` property in favor of passing in the `labware_id` into `__init__`. I wonder if we could check in with apps eng about if anyone ever uses `well.parent`?

It's also almost completely undocumented [in our own docs](https://docs.opentrons.com/v2/new_protocol_api.html#opentrons.protocol_api.labware.Well):

> _property_ **parent**
>
> _New in version 2.0._